clients/cli/src/prover.rs

```diff
@@ -157,22 +161,28 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         let start_time = Instant::now();
         let mut progress_time = start_time;
         for step in start..end {
-            proof = prove_seq_step(Some(proof), &pp, &tr).expect("error proving step");
+            // Generate proof for current step
+            proof =
+                prove_seq_step(Some(proof), &public_parameters, &tr).expect("error proving step");
             steps_proven += 1;
             completed_fraction = steps_proven as f32 / steps_to_prove as f32;
 
+            // Calculate performance metrics
             let progress_duration = progress_time.elapsed();
             let proof_cycles_hertz = k as f64 * 1000.0 / progress_duration.as_millis() as f64;
 
+            // Send progress update to orchestrator
             let progress = ClientProgramProofRequest {
```

Comment:
Pre-existing issue: I think we want to send this upon finishing a 10-step proof, rather than after every step.